ansible_events_ui/ruleset.py

```diff
@@ -56,57 +70,85 @@ async def activate_rulesets(
     Call ansible-events with ruleset, inventory, and extravars added
     as volumes to a container.
     """
-    tempdir = tempfile.mkdtemp(prefix="ruleset_manager")
-    result = await db.execute(select(playbooks))
-    for playbook in result.all():
-        with open(os.path.join(tempdir, playbook.name), "w") as f:
-            f.write(playbook.playbook)
-    rules_file = os.path.join(tempdir, "rules.yml")
-    with open(rules_file, "w") as f:
-        f.write(rulesets)
-    inventory_file = os.path.join(tempdir, "inventory.yml")
-    with open(inventory_file, "w") as f:
-        f.write(inventory)
-    vars_file = os.path.join(tempdir, "vars.yml")
-    with open(vars_file, "w") as f:
-        f.write(extravars)
+    settings = load_settings()
```

Comment:
You should not load settings second time here. Settings are loaded only once at the application initialization. 
Please use settings available via dependency injection.

Also the function shouldn't probably depend on a settings object at all, as it's refers to a specific parameter. You can pass the deployment type as a parameter explicitly.